File: microservices/claim-service/src/main/java/com/returnX/claim_service/service/ClaimService.java
package com.returnX.claim_service.service;

import com.returnX.claim_service.dto.ClaimResponse;
import com.returnX.claim_service.dto.CreateClaimResponse;

import java.util.List;

public interface ClaimService {

    CreateClaimResponse createClaim(Long itemId, String email);

    List<ClaimResponse> getMyClaims(String email);

    ClaimResponse getClaimById(Long claimId);

    ClaimResponse getByChatRoomId(String chatRoomId);
}

File: frontend/src/services/claimService.ts
import api from "./api";

export interface CreateClaimResponse {
  claimId: number;
  chatRoomId: string;
  message: string;
}

export interface ClaimResponse {
  id: number;
  itemId: number;
  ownerEmail: string;
  claimerEmail: string;
  chatRoomId: string;
  claimedAt: string;
}

export const createClaim = async (
  itemId: number
): Promise<CreateClaimResponse> => {
  const { data } = await api.post(`/claims/${itemId}`);
  return data;
};

export const getMyClaims = async (): Promise<ClaimResponse[]> => {
  const { data } = await api.get("/claims/my");
  return data;
};

export const getClaimById = async (
  claimId: number
): Promise<ClaimResponse> => {
  const { data } = await api.get(`/claims/${claimId}`);
  return data;
};

export const getOwnerClaims = async (): Promise<ClaimResponse[]> => {
  const { data } = await api.get("/claims/owner");
  return data;
};